src/test/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaRegexMapperTest.java

```diff
@@ -0,0 +1,245 @@
+     */
+    @Test
+    public void testDefaultTopic() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        MqttKafkaRegexMapper mapper = new MqttKafkaRegexMapper(rules);
+
+        assertThat("Should use the default topic when no mapping pattern matches.",
+                mapper.map("sensor/temperature").kafkaTopic(), is(MqttKafkaSimpleMapper.DEFAULT_KAFKA_TOPIC));
+
+        assertThat("The key for the default topic should be null",
+                mapper.map("sensor/temperature").kafkaKey(), nullValue());
+    }
+
+    /**
+     * Test the mapping of single level topics.
+     */
+    @Test
+    public void testSingleLevel() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        rules.add(new MappingRule("sensors/[^/]+/data", "sensor_data", null));
+        rules.add(new MappingRule("devices/([^/]+)/data", "devices_$1_data", "device_$1"));
+        rules.add(new MappingRule("fleet/([0-9]+)/vehicle/(\\w+)", "fleet_$1", "fleet$2"));
+        rules.add(new MappingRule("building/(\\d+)/floor/(\\d+).*", "building_$1_$2", "building_$1"));
+        rules.add(new MappingRule("term/(\\d+)", "term$1", "my_term_key_$1"));
+        rules.add(new MappingRule("devices/([^/]+)/data/(\\b(all|new|old)\\b)", "devices_$1_$2_data", "device_$1"));
+
+        MqttKafkaRegexMapper mapper = new MqttKafkaRegexMapper(rules);
+
+        assertThat("building/(\\d+)/floor/(\\d+).* should be mapped to building_$1_$2",
+                mapper.map("building/14/floor/25").kafkaTopic(), is("building_14_25"));
+
+        assertThat("The key for building_$1 should be expanded to building_14",
+                mapper.map("building/14/floor/25").kafkaKey(), is("building_14"));
+
+        assertThat("sensors/[^/]+/data should be mapped to sensor_data",
+                mapper.map("sensors/temperature/data").kafkaTopic(), is("sensor_data"));
+
+        assertThat("The key for the rule sensors/[^/]+/data should be null",
+                mapper.map("sensors/temperature/data").kafkaKey(), nullValue());
+
+        assertThat("devices/[^/]+/data should be mapped to devices_$1_data",
+                mapper.map("devices/123/data").kafkaTopic(), is("devices_123_data"));
+
+        assertThat("The key for devices/[^/]+/data should be device_$1",
+                mapper.map("devices/123/data").kafkaKey(), is("device_123"));
+
+        assertThat("fleet/([0-9]+)/vehicle/(\\w+) should be mapped to fleet_$1",
+                mapper.map("fleet/4/vehicle/23").kafkaTopic(), is("fleet_4"));
+
+        assertThat("The key for fleet/([0-9]+)/vehicle/(\\w+) should be fleet$2",
+                mapper.map("fleet/4/vehicle/23").kafkaKey(), is("fleet23"));
+
+        assertThat("term/(\\d+) should be mapped to term$1",
+                mapper.map("term/4").kafkaTopic(), is("term4"));
+
+        assertThat("The key for term/(\\d+) should be my_term_key_$1",
+                mapper.map("term/4").kafkaKey(), is("my_term_key_4"));
```

Comment:
For all the above you did what I already mentioned before. You should call `map` one time (for each pattern) and the run the asserts on the `MappingResult` instance.